src/probability/variance.lean

```diff
@@ -0,0 +1,226 @@
+/-
+Copyright (c) 2022 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+import probability.notation
+import probability.integration
+
+/-!
+# Variance of random variables
+
+We define the variance of a real-valued random variable as `Var[X] = 𝔼[(X - 𝔼[X])^2]` (in the
+`probability_theory` locale).
+
+We prove the basic properties of the variance:
+* `variance_le_expectation_sq`: the inequality `Var[X] ≤ 𝔼[X^2]`.
+* `meas_ge_le_mul_variance`: Chebyshev's inequality, i.e.,
+      `ℙ {ω | c ≤ |X ω - 𝔼[X]|} ≤ ennreal.of_real (Var[X] / c ^ 2)`.
+* `indep_fun.Var_add`: the variance of the sum of two independent random variables is the sum of
+  the variances.
+* `indep_fun.Var_sum`: the variance of a finite sum of pairwise independent random variables is
+  the sum of the variances.
+-/
+
+open measure_theory filter finset
+
+noncomputable theory
+
+open_locale big_operators measure_theory probability_theory ennreal nnreal
+
+namespace probability_theory
+
+/-- The variance of a random variable is `𝔼[X^2] - 𝔼[X]^2` or, equivalently, `𝔼[(X - 𝔼[X])^2]`. We
+use the latter as the definition, to ensure better behavior even in garbage situations. -/
+def variance {Ω : Type*} {m : measurable_space Ω} (f : Ω → ℝ) (μ : measure Ω) :=
+μ[(f - (λ x, μ[f])) ^ 2]
```

Comment:
Could we allow non-real random variables with `μ[∥f - (λ x, μ[f])∥ ^ 2]` as the definition? This will useful if we want to work with complex random variables.

Reply:
For vector valued random variables, the right notion is the covariance. It is indeed useful, but quite different, so I'd rather stick to the real-valued case for the variance.